### PR TITLE
CIRC-10145 OpenTelemetry Protobuf GRPC Support

### PR DESCRIPTION
**Pull request by vasuraman11**
https://circonus.atlassian.net/browse/CIRC-10145

The OLTPHTTP module has been updated to enable usage of GRPC as the transport protocol for OpenTelemetry protobuf metric data.  This is dependent on and requires GRPC 1.55 and Protobuf 3.23, as well as a patched Protobuf-C that can be built against that version of Protobuf.

### CODE REVIEW

File: buildtools/u2204/env-asan.inc [hunk 1]
```diff
@@ -0,0 +1,19 @@
+source buildtools/u2004/env.inc
+export CFLAGS="\
+ -Wall\
+ -Werror\
+ -Wno-deprecated-declarations\
+ -Wno-unused-variable\
+ -Wno-unused-function\
+ -g3\
+ -ggdb3\
```
Comment (pjulien) [inline buildtools/u2204/env-asan.inc:9]:
```suggestion
```

File: buildtools/u2204/env-asan.inc [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ -fsanitize=address\
+ -fsanitize-recover=address"
+export CXXFLAGS="$CXXFLAGS -g3 -ggdb3 -fsanitize=address -fsanitize-recover=address"
```
Comment (pjulien) [inline buildtools/u2204/env-asan.inc:12]:
```suggestion
export CXXFLAGS="$CXXFLAGS -g3 -fsanitize=address -fsanitize-recover=address"
```

File: buildtools/u2204/env.inc [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ -Werror\
+ -Wno-deprecated-declarations\
+ -Wno-unused-variable\
+ -Wno-format-truncation\
+ -Wno-misleading-indentation\
+ -Wno-stringop-overflow"
+export CXXFLAGS="\
+ -ggdb3\
+ -O2\
+ -Wall\
```
Comment (pjulien) [inline buildtools/u2204/env.inc:29]: Please select a C++ standard

File: buildtools/u2204/env.inc [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+export CFLAGS="\
+ -ggdb3\
+ -O2\
+ -Wall\
+ -Werror\
+ -Wno-deprecated-declarations\
+ -Wno-unused-variable\
+ -Wno-format-truncation\
+ -Wno-misleading-indentation\
+ -Wno-stringop-overflow"
```
Comment (pjulien) [inline buildtools/u2204/env.inc:25]: Please select a C standard

File: buildtools/u2204/env.inc [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+:/usr/bin\
+:/sbin\
+:/bin"
+export CPPFLAGS="\
+ -I/opt/circonus/include\
+ -I/opt/circonus/include/luajit\
+ -I/opt/circonus/include/mysql\
+ -DGTEST_HAS_PTHREAD=1"
+export CFLAGS="\
+ -ggdb3\
```
Comment (pjulien) [inline buildtools/u2204/env.inc:17]:
```suggestion
 -g3\
```

File: buildtools/u2204/env.inc [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+ -O2\
+ -Wall\
+ -Werror\
+ -Wno-deprecated-declarations\
+ -Wno-unused-variable\
+ -Wno-format-truncation\
+ -Wno-misleading-indentation\
+ -Wno-stringop-overflow"
+export CXXFLAGS="\
+ -ggdb3\
```
Comment (pjulien) [inline buildtools/u2204/env.inc:27]:
```suggestion
 -g3 \
```

File: src/libnoit.c [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+ *       copyright notice, this list of conditions and the following
+ *       disclaimer in the documentation and/or other materials provided
+ *       with the distribution.
+ *     * Neither the name OmniTI Computer Consulting, Inc. nor the names
+ *       of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written
+ *       permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
```
Comment (pjulien) [inline src/libnoit.c:20]: this isn't part of your commit, please rework your PR so it's rebased on the tip of master

File: buildtools/u2004/env.inc [hunk 8]
```diff
@@ -25,7 +25,8 @@ export CXXFLAGS="\
  -ggdb3\
```
Comment (pjulien) [inline buildtools/u2004/env.inc]:
```suggestion
 -g3 \
```

File: src/modules/Makefile.in [hunk 9]
```diff
@@ -25,7 +25,7 @@ MODULELD=@MODULELD@
 MODULECC=@MODULECC@
 MODULESHCFLAGS=@MODULESHCFLAGS@
 MODULECXX=@MODULECXX@
-MODULESHCXXFLAGS=@MODULESHCXXFLAGS@ -D__STDC_LIMIT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_CONSTANT_MACROS -std=c++17
+MODULESHCXXFLAGS=@MODULESHCXXFLAGS@ -Wno-mismatched-new-delete -Wno-sign-compare -D__STDC_LIMIT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_CONSTANT_MACROS -DPROTOBUF_USE_DLLS -std=c++17
```
Comment (pjulien) [inline src/modules/Makefile.in:28]:
```suggestion
MODULESHCXXFLAGS=@MODULESHCXXFLAGS@ -Wno-mismatched-new-delete -Wno-sign-compare -D__STDC_LIMIT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_CONSTANT_MACROS -DPROTOBUF_USE_DLLS -std=c++20
```

File: buildtools/u2204/env-ubsan.inc [hunk 10]
```diff
@@ -0,0 +1,18 @@
+source buildtools/u2004/env.inc
+export CFLAGS="\
+ -Wall\
+ -Werror\
+ -Wno-deprecated-declarations\
+ -Wno-unused-variable\
+ -Wno-unused-function\
+ -g3\
+ -ggdb3\
```
[the remark this answers is not in the document]
Comment (vasuraman11) [reply buildtools/u2204/env-ubsan.inc]: not part of this PR (was divergent commits, have been resolved)

File: buildtools/u2204/env-ubsan.inc [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+ -fno-omit-frame-pointer\
+ -fsanitize=undefined"
+export CXXFLAGS="$CXXFLAGS -g3 -ggdb3 -fno-omit-frame-pointer -fsanitize=undefined"
```
[the remark this answers is not in the document]
Comment (vasuraman11) [reply buildtools/u2204/env-ubsan.inc]: not part of this PR (was divergent commits, have been resolved)

File: buildtools/u2204/env-asan.inc (hunk 2 above)
Comment (vasuraman11) [reply buildtools/u2204/env-asan.inc]: not part of this PR (was divergent commits, have been resolved)

File: buildtools/u2204/env.inc (hunk 3 above)
Comment (vasuraman11) [reply buildtools/u2204/env.inc]: not part of this PR (was divergent commits, have been resolved)

File: buildtools/u2204/env.inc (hunk 4 above)
Comment (vasuraman11) [reply buildtools/u2204/env.inc]: not part of this PR (was divergent commits, have been resolved)

File: buildtools/u2204/env.inc (hunk 6 above)
Comment (vasuraman11) [reply buildtools/u2204/env.inc]: not part of this PR (was divergent commits, have been resolved)

File: src/modules/Makefile.in (hunk 9 above)
Comment (vasuraman11) [reply src/modules/Makefile.in]: tested, merged

**Review by pjulien — APPROVED** [review]

**Review by pjulien — APPROVED** [review]

File: src/Makefile.in [hunk 12]
```diff
@@ -484,10 +484,10 @@ Makefile.dep:
 	$(top_srcdir)/buildtools/culldeps.sh Makefile.dep
 
 js/bundle.desc:	proto/bundle.proto
-	protoc -ojs/bundle.desc proto/bundle.proto
+	LD_LIBRARY_PATH=/opt/circonus/lib protoc -ojs/bundle.desc proto/bundle.proto
 
 bundle.pb-c.c bundle.pb-c.h: proto/bundle.proto
-	(cd proto && protoc bundle.proto --java_out=`pwd`/../java && protoc-c bundle.proto --c_out=`pwd`/../ && cd ..)
+	(cd proto && LD_LIBRARY_PATH=/opt/circonus/lib protoc bundle.proto --java_out=`pwd`/../java && LD_LIBRARY_PATH=/opt/circonus/lib protoc-c bundle.proto --c_out=`pwd`/../ && cd ..)
```
Comment (esproul) [inline src/Makefile.in:490]: It is not safe to assume `/opt/circonus/lib` for these locations. A properly built set of dependencies should not require this.

File: src/modules/Makefile.in [hunk 13]
```diff
@@ -25,7 +25,7 @@ MODULELD=@MODULELD@
 MODULECC=@MODULECC@
 MODULESHCFLAGS=@MODULESHCFLAGS@
 MODULECXX=@MODULECXX@
-MODULESHCXXFLAGS=@MODULESHCXXFLAGS@ -D__STDC_LIMIT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_CONSTANT_MACROS -std=c++17
+MODULESHCXXFLAGS=@MODULESHCXXFLAGS@ -Wno-mismatched-new-delete -Wno-sign-compare -D__STDC_LIMIT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_CONSTANT_MACROS -DPROTOBUF_USE_DLLS -std=c++20
```
Comment (esproul) [inline src/modules/Makefile.in:28]: What code triggers these warnings, and if it's ours, can we not fix it so these suppressions are unnecessary?

Comment (vasuraman11) [reply src/modules/Makefile.in:28]: its in 3rd party code grpc/protobuf, have to live with it for now

File: src/Makefile.in (hunk 12 above)
Comment (vasuraman11) [reply src/Makefile.in:490]: fixed

Comment (vasuraman11) [reply src/Makefile.in:490]: this was residue from before you fixed the issue

**Review by esproul — APPROVED** [review]
LGTM

**Pull request merged**